scripts/xtensa-build-all.sh

```diff
@@ -30,30 +33,71 @@ usage: xtensa-build.sh [options] platform(s)
        -u Force UP ARCH
        -d Enable debug build
        -c Interactive menuconfig
+       -o override config
        -k Use private key
        -v Verbose Makefile log
        -j n Set number of make build jobs. Jobs=#cores when no flag. \
 Infinte when not specified.
+	-m MEU path
+	-s openssl path
        Supported platforms ${SUPPORTED_PLATFORMS[*]}
 EOF
 }
 
+if [ $# -eq 0 ]
+then
+	print_usage
+	exit 1
+fi
+
 # parse the args
-while getopts "rudj:ckva" OPTION; do
+while getopts "rudj:ck:vao:m:s:" OPTION; do
         case "$OPTION" in
 		r) BUILD_ROM=yes ;;
 		u) BUILD_FORCE_UP=yes ;;
 		d) BUILD_DEBUG=yes ;;
 		j) BUILD_JOBS=$OPTARG ;;
 		c) MAKE_MENUCONFIG=yes ;;
-		k) USE_PRIVATE_KEY=yes ;;
+		o) OVERRIDE_CONFIG=$OPTARG ;;
+		k) PRIVATE_KEY=$OPTARG ;;
 		v) BUILD_VERBOSE='VERBOSE=1' ;;
 		a) PLATFORMS=("${SUPPORTED_PLATFORMS[@]}") ;;
+		m) MEU_TOOL_PATH=$OPTARG ;;
+		s) MEU_OPEN_SSL_PATH=$OPTARG ;;
 		*) print_usage; exit 1 ;;
         esac
 done
 shift $((OPTIND-1))
 
+SIGNING_TOOL=$(echo $SIGNING_TOOL | tr '[:upper:]' '[:lower:]')
+
+if [ -n "${OVERRIDE_CONFIG}" ]
+then
+	OVERRIDE_CONFIG="src/arch/xtensa/configs/override/$OVERRIDE_CONFIG.config"
```

Comment:
> This assume the scritpts can only be called with scripts/xtensa-build-all.sh

Good point but this is already the case, so adding support here could not even be tested without fixing first similar issues in a number of other places in the script: "yak shaving".

It is already the case because `cmake` points at the source directory with `..` and probably other places too: for instance the `pwd` dance at the top seems very suspicious.